unstable/wlr-export-dmabuf-unstable-v1.xml

```diff
@@ -0,0 +1,244 @@
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+  </copyright>
+
+  <description summary="a protocol for low overhead screen content capturing">
+    An interface to capture surfaces in an efficient way.
+    Overall usage:
+
+    1.) client registers with zwlr_screencontent_manager_v1
+    2.) server sends client info about surfaces via "receive_surface_info"
+    3.) client subscribes to capture a surface via the "capture" requests
+    4.) server sends client events via the "zwlr_screencontent_frame" interface
+    5.) client finishes and informs server via the "frame_destroy" event
+    6.) client optionally resubscribes via repeating steps 3.) through 5.)
+  </description>
+
+  <interface name="zwlr_export_dmabuf_frame_v1" version="1">
+    <description summary="a frame ready for readout">
+      This object represents a frame which is ready to have its resources
+      fetched and used.
+
+      The receive callback shall be called first, followed by the "object"
+      callback once per dmabuf object or the "plane" callback, once per dmabuf
+      plane. The "ready" event is called last to indicate that all the data has
+      been made available for readout, as well as the time at which presentation
+      happened at. The ownership of the frame is passed to the client, who's
+      responsible for destroying it via the "destroy" event once finished and
+      by calling close() on the file descriptors received.
+      All frames are read-only and may not be written into or altered.
+    </description>
+
+    <enum name="flags">
+      <description summary="frame flags">
+        Special flags that should be respected by the client.
+      </description>
+      <entry name="transient" value="0x1" since="1"
+             summary="clients should copy frame before processing"/>
+    </enum>
+
+    <event name="frame">
+      <description summary="main callback">
+        Main callback supplying the client with information about the frame,
+        as well as an object to serve as context for destruction. Always called
+        first before any other events.
+
+        The "transform" argument describes the orientation needed to be applied
+        to correctly orient the buffer. For example, a buffer rotated by 90
+        degrees will have a value of "3" here, corresponding to the need to
+        apply a 270 degree transpose to correctly present the buffer.
+      </description>
+      <arg name="width" type="uint"
+           summary="frame width, scaling factor included"/>
+      <arg name="height" type="uint"
+           summary="frame height, scaling factor included"/>
+      <arg name="offset_x" type="uint"
+           summary="crop offset for the x axis"/>
+      <arg name="offset_y" type="uint"
+           summary="crop offset for the y axis"/>
```

Comment:
Security-wise, there's no way of enforcing these offsets?

Reply:
Clients should sanity check when it makes sense (you've downloaded the frame and now you have to crop it) or do nothing (you're rendering the frame to some framebuffer and all that can happen is its not going to be visible). Its the same as any other part of the protocol (check the fds, check the format, check the modifier, etc.).

Reply:
Nah, I mean: if the client asks to capture another client, and the compositor decides to crop the DMA-BUF of the whole screen with these offsets, the client can ignore those and read pixels it shouldn't have access to.